js/components/Current.js

```diff
@@ -0,0 +1,82 @@
+import { elements } from './../vendor/elements';
+
+export default class Current {
+
+  constructor(props) {
+    this.props = props;
+    this.state = null;
+  }
+
+  update(nextProps) {
+    this.props = Object.assign({}, this.props, nextProps);
+    this.updateState({current: this.props.current});
+  }
+
+  updateState(nextState) {
```

Comment:
Same story - you should move `updateState` to the parent class and then just reuse thru its ancestors.